src/lib.rs

```diff
@@ -3416,6 +3429,11 @@ impl Connection {
             self.max_rx_data_next / 2 > self.max_rx_data - self.rx_data
     }
 
+    /// Commits the new max_rx_data limit.
+    pub fn update_max_data(&mut self) {
+        self.max_rx_data = self.max_rx_data_next;
```

Comment:
Nitpick, but do we really need a separate method for this? It's only called once and the field is not hidden so can just move this line to where `self.almost_full = false;` is.

Reply:
Ok. I inlined the function to where we send MAX_DATA.